### PR TITLE
fixed decode token issue

### PR DESCRIPTION
**Pull request by krezzoid**
Hey,

I found strange issue, with used `res.content.decode(Result.self)` access_token was not decoded (at the same time JSON body with access_token received) and in total it was impossible to send FCM message.

So, I replaced this call to decode access_token using JSON decoder and everything works well.

### CODE REVIEW

File: Sources/FCM/Helpers/FCM+AccessToken.swift [hunk 1]
```diff
@@ -19,9 +19,14 @@ extension FCM {
         .validate()
         .flatMapThrowing { res -> String in
             struct Result: Codable {
-                var access_token: String
+                let access_token: String
             }
-            let result = try res.content.decode(Result.self)
```
Comment (MihaelIsaev) [inline Sources/FCM/Helpers/FCM+AccessToken.swift:24]:
```swift
guard let body = res.body else {
    throw Abort(.notFound, reason: "Data not found")
}
let result = try JSONDecoder().decode(Result.self, from: res.body)
```
Could you please try this way? NIO have an extension for `JSONEncoder` with byteBuffer
Screenshot 2020-09-15 at 19 47 17

**Review by MihaelIsaev — CHANGES_REQUESTED** [review]

File: Sources/FCM/Helpers/FCM+AccessToken.swift (hunk 1 above)
Comment (MihaelIsaev) [reply Sources/FCM/Helpers/FCM+AccessToken.swift:24]: Or even this way would be better 🙂 
```swift
let result = try res.content.decode(Result.self, using: JSONDecoder())
```

Comment (krezzoid) [reply Sources/FCM/Helpers/FCM+AccessToken.swift:24]: yep, right

**Review by MihaelIsaev — APPROVED** [review]
Thank you for the investigation and fix ✈️😊🥳

**Comment by MihaelIsaev** [comment]
Thank you! It is available in 2.8.1 now 🚀

**Pull request merged**